src/core/utils.cc

```diff
@@ -37,6 +38,252 @@
 
 namespace nvidia { namespace inferenceserver {
 
+namespace {
```

Comment:
These first set of functions should go in a new file, called "ensemble_utils.cc". Can you rename utils -> model_config_utils. In model_config_utils you should have the validation for a model config that can be performed in isolation (with just the one config). So you can have some validation on the structure of the Step inputs and outputs and their naming, etc. For the ensemble_utils routines those can only be used once all model configs are loaded and will need to be re-run whenever models are loaded and unloaded... That will probably have to happen in the ModelConfigManager. Not needed for this change since you can call directly in the unit tests, but something to think about.